### DISCUSSION TITLE
Import variables to scss files

### DISCUSSION THREAD

**Discussion by felixboet** [vercel/next.js#19672] [Help] [answered]
Hey there. Any idea how to import variables from a json or md file to sass in next.js? I thought about using the npm module [node-sass-json-importer](https://github.com/pmowrer/node-sass-json-importer), but not sure if/how its possible to hook into node-sass in next.js. Or maybe there is another solution... I am very thankful for any suggestion or hint :) Felix

**Comment by Tomburgs** [comment, marked as answer]
You could use css variables & define them in a style tag in `_app.tsx`

Something like so:
```jsx
const myVariable = '50px';
const myStyles = `
  :root {
    --my-variable: ${myVariable};
  }
`;

...
<style>{ myStyles }</style>
```

Then from within CSS you can just do
```css
p {
  font-size: var(--my-variable);
}
```

**Reply by felixboet** [reply]
Thanks @Tomburgs ! ... Nice, but I think it wouldn't solve my problem. I wanted to parse variables into a SCSS-file. To be more specific: I want to use bootstrap as css framework and be able to set bootstrap sass variables like "$body-color" from my CMS, so I need to parse variables from a json/markdown-file into the SCSS before compiling to css

**Reply by Tomburgs** [reply]
Understood, in that case you just want to add `node-sass-json-importer` to your `next.config.js` like so:
```js
const jsonImporter = require('node-sass-json-importer');

module.exports = {
  ...
  sassOptions: {
    importer: jsonImporter()
  }
  ...
};
```

I'm not familiar with how to correctly configure this library, but the importer will get called as expected 🎉.

**Reply by felixboet** [reply]
Thanks 🥇 Worked exactly as @Tomburgs proposed:
```
const jsonImporter = require("node-sass-json-importer");

module.exports = {
  sassOptions: {
    importer: jsonImporter(),
  },
};
```
then you can use something like `@import "variables.json";`  in your scss files

**Comment by susanta96** [comment]
You could use [Json2scss-map](https://npmjs.com/package/json2scss-map).
